src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/Constants.cs

```diff
@@ -3,34 +3,40 @@
 
 using System.Diagnostics;
 using System.Linq;
-using System.Net.Http.Headers;
 using System.Reflection;
+using System.Runtime.InteropServices;
 
 namespace Microsoft.AspNetCore.Http.Connections.Client.Internal
 {
     internal static class Constants
     {
-        public static readonly ProductInfoHeaderValue UserAgentHeader;
+        public const string UserAgent = "User-Agent";
+        public static readonly string UserAgentHeader;
 
         static Constants()
         {
-            var userAgent = "Microsoft.AspNetCore.Http.Connections.Client";
-
             var assemblyVersion = typeof(Constants)
                 .Assembly
                 .GetCustomAttributes<AssemblyInformationalVersionAttribute>()
                 .FirstOrDefault();
 
             Debug.Assert(assemblyVersion != null);
 
+            var majorVersion = typeof(Constants).Assembly.GetName().Version.Major;
+            var minorVersion = typeof(Constants).Assembly.GetName().Version.Minor;
+            var os = RuntimeInformation.OSDescription;
+            var runtime = ".NET";
+            var runtimeVersion = RuntimeInformation.FrameworkDescription;
+
             // assembly version attribute should always be present
             // but in case it isn't then don't include version in user-agent
             if (assemblyVersion != null)
             {
-                userAgent += "/" + assemblyVersion.InformationalVersion;
+                UserAgentHeader = $"Microsoft SignalR/{majorVersion}.{minorVersion} ({assemblyVersion.InformationalVersion}; {os}; {runtime}; {runtimeVersion})";
+            } else
+            {
+                UserAgentHeader = $"Microsoft SignalR/{majorVersion}.{minorVersion}; {os}; {runtime}; {runtimeVersion})";
```

Comment:
```suggestion
                UserAgentHeader = $"Microsoft SignalR/{majorVersion}.{minorVersion}; ({os}; {runtime}; {runtimeVersion})";
```

Reply:
I wish there was a good way to test this branch, but I'm not sure there is.